sklearn/random_projection.py

```diff
@@ -368,26 +378,15 @@ def fit(self, X, y=None):
                 n_samples=n_samples, eps=self.eps
             )
 
-            if self.n_components_ <= 0:
-                raise ValueError(
-                    "eps=%f and n_samples=%d lead to a target dimension of "
-                    "%d which is invalid" % (self.eps, n_samples, self.n_components_)
-                )
-
-            elif self.n_components_ > n_features:
+            if self.n_components_ > n_features:
```

Comment:
This validation is on `self.n_components_` which is the fitted attribute, **not** the given parameter `self.n_components`. Indeed `self.n_components_ = johnson_lindenstrauss_min_dim(...)` so this validates the output to that function and is still necessary. Reverting this should be sufficient.

Reply:
Yeah, I was sloppy!